src/firefox/api.js

```diff
@@ -304,6 +304,19 @@ this.experiments_firefox_bridge = class extends ExtensionAPI {
             let win = Services.wm.getMostRecentWindow("navigator:browser");
             win.BrowserOpenAddonsMgr("addons://shortcuts/shortcuts");
           },
+
+          /**
+           * Gets the telemetry ID for the Firefox profile to link to the extension.
+           *
+           * @returns {Promise<string>} The telemetry ID
+           */
+          async getTelemetryID() {
+            let telemetryID = Services.prefs.getStringPref(
+              "browser.firefoxbridge.installId",
+              undefined,
```

Comment:
Is it better to make the default ""? Just so that anything that expects a valid string works?

Reply:
Undefined would prevent the ping from being sent, whereas "" would send the telemetry ping with an empty string. I suppose "" would let us see which users failed to retrieve the ID.

Reply:
> Undefined would prevent the ping from being sent,

... due to an error being thrown by getStringPref, right?

> whereas "" would send the telemetry ping with an empty string. I suppose "" would let us see which users failed to retrieve the ID.

`""` would make more sense here. Then at the caller you could check whether the return value is falsey/`""` and return or throw if desired.

Reply:
It would fail during the glean ping as it expects a string value but receives undefined, getStringPref doesn't fail with the default value set.

But yes, I'll change this to "" and handle that